src/Instance.cs

```diff
@@ -579,6 +580,95 @@ public Instance(Store store, Module module, params object[] imports)
             return _store.GetCachedExtern(ext.of.global);
         }
 
+        /// <summary>
+        /// Get all exported functions
+        /// </summary>
+        /// <returns>An enumerable of functions exported from this instance</returns>
+        public IEnumerable<(string Name, Function Function)> GetFunctions()
+        {
+            for (var i = 0; i < int.MaxValue; i++)
+            {
+                if (TryGetExtern(i, ExternKind.Func) is not var (name, @extern))
+                    break;
+
+                yield return (name, _store.GetCachedExtern(@extern.of.func));
+            }
+
+            GC.KeepAlive(_store);
+        }
+
+        /// <summary>
+        /// Get all exported tables
+        /// </summary>
+        /// <returns>An enumerable of tables exported from this instance</returns>
+        public IEnumerable<(string Name, Table Table)> GetTables()
+        {
+            for (var i = 0; i < int.MaxValue; i++)
+            {
+                if (TryGetExtern(i, ExternKind.Table) is not var (name, @extern))
+                    break;
+
+                yield return (name, new Table(_store, @extern.of.table));
+            }
+
+            GC.KeepAlive(_store);
+        }
+
+        /// <summary>
+        /// Get all exported memories
+        /// </summary>
+        /// <returns>An enumerable of memories exported from this instance</returns>
+        public IEnumerable<(string Name, Memory Memory)> GetMemories()
+        {
+            for (var i = 0; i < int.MaxValue; i++)
+            {
+                if (TryGetExtern(i, ExternKind.Memory) is not var (name, @extern))
+                    break;
+
+                yield return (name, _store.GetCachedExtern(@extern.of.memory));
+            }
+
+            GC.KeepAlive(_store);
+        }
+
+        /// <summary>
+        /// Get all exported globals
+        /// </summary>
+        /// <returns>An enumerable of globals exported from this instance</returns>
+        public IEnumerable<(string Name, Global Global)> GetGlobals()
+        {
+            for (var i = 0; i < int.MaxValue; i++)
+            {
+                if (TryGetExtern(i, ExternKind.Global) is not var (name, @extern))
+                    break;
+
+                yield return (name, _store.GetCachedExtern(@extern.of.global));
+            }
+
+            GC.KeepAlive(_store);
+        }
+
+        /// <summary>
+        /// Try to get the nth extern, optionally filtered to a specific type
+        /// </summary>
+        /// <param name="i"></param>
+        /// <param name="type"></param>
+        /// <returns></returns>
```

Comment:
Nit: we should probably either fill this all out or remove the comment entirely as this is a private function anyway (and the purpose of which is fairly evident in the name).